src/app/segmentation/page.tsx

```diff
@@ -1,23 +1,86 @@
-import React from 'react';
+'use client';
+
+import React, { useState } from 'react';
 import dynamic from 'next/dynamic';
 import './segmentation.css';
 import ApproveSegButton from './approveSeg';
 import NavBar from '../navbar/navbar';
-const VisualizationComponent = dynamic(
-  () => import('../services/visualizationCompare'),
+import * as CONSTANTS from '../constants';
+
+const VisualizationComponentT1C = dynamic(
+  () => import('../services/visualizationCompareT1C'),
+  {
+    ssr: false,
+  }
+);
+
+const VisualizationComponentT1N = dynamic(
+  () => import('../services/visualizationCompareT1N'),
+  {
+    ssr: false,
+  }
+);
+
+const VisualizationComponentT2F = dynamic(
+  () => import('../services/visualizationCompareT2F'),
+  {
+    ssr: false,
+  }
+);
+
+const VisualizationComponentT2W = dynamic(
+  () => import('../services/visualizationCompareT2W'),
   {
     ssr: false,
   }
 );
 
 export default function Page() {
+  const [modality, setModality] = useState(CONSTANTS.T2F);
+
+  const prevModality = () => {
+    if (modality == CONSTANTS.T1C) {
+      setModality(CONSTANTS.T2W);
+    } else if (modality == CONSTANTS.T1N) {
+      setModality(CONSTANTS.T1C);
+    } else if (modality == CONSTANTS.T2F) {
+      setModality(CONSTANTS.T1N);
+    } else if (modality == CONSTANTS.T2W) {
+      setModality(CONSTANTS.T2F);
+    }
+  };
+  
+  const nextModality = () => {
+    if (modality == CONSTANTS.T1C) {
+      setModality(CONSTANTS.T1N);
+    } else if (modality == CONSTANTS.T1N) {
+      setModality(CONSTANTS.T2F);
+    } else if (modality == CONSTANTS.T2F) {
+      setModality(CONSTANTS.T2W);
+    } else if (modality == CONSTANTS.T2W) {
+      setModality(CONSTANTS.T1C);
+    }
+  };
+
   return <>
     <NavBar view="segmentation" />
     <h1>Approve Segmentation Images</h1>
+    <h2>Modality: {modality}</h2>
     <div className='flex justify-center mb-7' id='seg-preview'>
```

Comment:
Note: I wanted to use one visualization component with modality param, but render wouldn't update on modality change; needed to use separate visualizations for each modality for it to work.

Reply:
we might wanna add some loading later on so the page glitching thing isnt as bad (maybe some opaque box on top while loading happens in the back idk

Reply:
good point, a later task it is